src/components/Select.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  useState,
+  useEffect,
+  useRef,
+  ReactNode,
+  SelectHTMLAttributes,
+} from "react";
+
+import { DropdownDown, DropdownUp } from "@/assets/icon";
+import { cn } from "@/utils/cn";
+
+const sizeMap = {
+  lg: "py-4 px-5 text-[1rem]",
+  sm: "p-2.5 text-sm",
+} as const;
+
+interface Option {
+  label: string;
+  value: string;
+}
+
+interface SelectProps
+  extends Omit<
+    SelectHTMLAttributes<HTMLButtonElement>,
+    "size" | "onChange" | "disabled"
+  > {
+  id?: string;
+  label?: string;
+  options: Option[];
+  value?: string;
+  onValueChange?: (value: string) => void;
+  placeholder?: string;
+  size?: keyof typeof sizeMap;
+  fullWidth?: boolean;
+  width?: number | string;
+  className?: string;
+  wrapperClassName?: string;
+}
+
+// 라벨과 입력 영역을 감싸는 공통 컴포넌트
+function Field({
+  id,
+  label,
+  children,
+}: {
+  id?: string;
+  label?: string;
+  children: ReactNode;
+}) {
+  return (
+    <div>
+      {label && (
+        <label htmlFor={id} className="inline-block mb-2 leading-[1.625rem]">
+          {label}
+        </label>
+      )}
+      {children}
+    </div>
+  );
+}
+
+function Select({
+  id,
+  label,
+  options,
+  value,
+  onValueChange,
+  placeholder = "선택",
+  size = "lg",
+  fullWidth,
+  width,
+  className,
+  wrapperClassName,
+  ...rest
+}: SelectProps) {
+  const [open, setOpen] = useState(false);
+  const [buttonWidth, setButtonWidth] = useState<number>(0);
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  const selectedOption = options.find((option) => option.value === value);
+
+  const wrapperClassNames = cn(
+    "relative",
+    {
+      "w-full": fullWidth,
+    },
+    wrapperClassName,
+  );
+
+  const buttonClassNames = cn(
+    "flex items-center justify-between rounded-[0.375rem]",
+    {
+      "w-full": fullWidth,
+      "bg-white placeholder:text-gray-40 border border-gray-30": size === "lg",
+      "bg-gray-10 font-bold": size === "sm",
+    },
+    sizeMap[size],
+    className,
+  );
```

Comment:
`span` 태그를 사용하지 않고, `button` 태그의 `className`속성으로만 `placeholder`의 스타일을 제어하는 건 어떨까요? 🤔
그리고 `value`가 존재한다면, `text-black`이 적용 되어야 합니다. 😅

```suggestion
  const buttonClassNames = cn(
    "flex items-center justify-between rounded-[0.375rem]",
    {
      "w-full": fullWidth,
      "bg-white placeholder:text-gray-40 border border-gray-30": size === "lg",
      "bg-gray-10 font-bold": size === "sm",
    },
    value ? "text-black" : "text-gray-40", // 추가
    sizeMap[size],
    className,
  );
```
```tsx
<button
  id={id}
  type="button"
  ref={buttonRef}
  onClick={() => setOpen((prev) => !prev)}
  className={buttonClassNames}
  style={width ? { width } : undefined}
  {...rest}
>
  {/* <span className={cn(value ? "" : "text-gray-40")}> */}
    {selectedOption?.label || placeholder}
  {/* </span> */}
  {open ? <DropdownUp className="ml-2" /> : <DropdownDown className="ml-2" />}
</button>
```